### PR TITLE
feat: update step

### PR DESCRIPTION
**Pull request by zkwolf**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/steps/index.jsx [hunk 1]
```diff
@@ -41,30 +43,27 @@ const Steps = {
...
-        prefixCls,
-        ...props,
-      },
-      on: getListeners(this),
-      scopedSlots: this.$scopedSlots,
+      icons,
+      iconPrefix,
+      prefixCls,
+      progressDot,
+      ...props,
```
Comment (Amour1688) [inline components/steps/index.jsx:57]: 需要传入 `$attrs`，props 里面默认没有 event

File: components/_util/props-util.js [hunk 2]
```diff
@@ -268,15 +269,22 @@ export function getComponentName(opts) {
   return opts && (opts.Ctor.options.name || opts.tag);
 }
 
+export function isFragment(c) {
+  return c.length === 1 && c[0].type === Fragment;
+}
+
 export function isEmptyElement(c) {
-  return typeof c.type === 'symbol' && c.children.trim() === '';
+  return typeof c.type === Comment;
```
[the remark this answers is not in the document]
Comment (zkwolf) [reply components/_util/props-util.js]: ok

File: components/steps/index.jsx [hunk 3]
```diff
@@ -41,30 +39,27 @@ const Steps = {
     const getPrefixCls = this.configProvider.getPrefixCls;
     const prefixCls = getPrefixCls('steps', customizePrefixCls);
     const iconPrefix = getPrefixCls('', customizeIconPrefixCls);
+    const progressDot = this.progressDot || this.$slots.progressDot;
```
Comment (tangjinzhou) [inline components/steps/index.jsx:42]: getComponent(this, 'progressDot')

File: components/vc-steps/Step.jsx [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
-import { getOptionProps, getComponentFromProp, getListeners } from '../_util/props-util';
+import { getOptionProps, getComponent } from '../_util/props-util';
 
 function isString(str) {
   return typeof str === 'string';
 }
 function noop() {}
 export default {
   name: 'Step',
+  inheritAttrs: false,
```
Comment (tangjinzhou) [inline components/vc-steps/Step.jsx:10]: 如果声明了 inheritAttrs: false, class style 都需要单独处理
如果不声明就要单独声明事件

File: components/vc-steps/Steps.jsx [hunk 5]
```diff
@@ -119,42 +117,36 @@ export default {
...
-            on: getEvents(child),
-            scopedSlots: $scopedSlots,
+            stepNumber: `${stepNumber + 1}`,
+            stepIndex: stepNumber,
+            prefixCls,
+            iconPrefix,
+            progressDot,
+            icons,
+            ...childProps,
+            ...this.$attrs,
```
Comment (tangjinzhou) [inline components/vc-steps/Steps.jsx:137]: 这里只需要传 this.$attrs.slots 就可以了吧   多测试下   单测可以改一下了

Comment (zkwolf) [reply components/vc-steps/Steps.jsx:137]: 我先看看test-utils都改啥了，目前steps里面除了demo和mountTest好像没啥单测了

Comment (zkwolf) [reply components/vc-steps/Steps.jsx:137]: 这里我想了一下不用传this.$attrs了，里面没有需要从父组件传子组件的，progressDot已经从props传了

File: components/vc-steps/Steps.jsx [hunk 6]
```diff
@@ -119,42 +117,36 @@ export default {
...
     const stepsProps = {
       class: classString,
       ref: 'vcStepsRef',
-      on: listeners,
     };
     return (
       <div {...stepsProps}>
         {filteredChildren.map((child, index) => {
-          const childProps = getPropsData(child);
+          const childProps = child.props || {};
```
Comment (tangjinzhou) [inline components/vc-steps/Steps.jsx:127]: 还是用 getPropsData   不然有驼峰和下划线问题

**Pull request merged**